src/components/layout/AddToWishlistButton/AddToWishlistButton.js

```diff
@@ -4,49 +4,39 @@ import axiosInstance from '../../../apis/axiosClient'; // axios instance
 import { AuthStatus, useAuth } from '@akosasante/react-auth-context';
 import { useNavigate } from 'react-router-dom';
 
-const wishlistEndpoint = '/wishlist';
-
-const AddToWishlistButton = ({ album }) => {
+const AddToWishlistButton = ({ album, wishListId }) => {
   const [isAdded, setIsAdded] = useState(false);
   const { status } = useAuth();
   const navigate = useNavigate();
   const isUserLoggedIn = status === AuthStatus.LoggedIn;
 
-  const fetchOrCreateWishlist = async () => {
-    try {
-      const response = await axiosInstance.post(wishlistEndpoint);
-      return response.data.wishlist._id; // Return the wishlist ID
-    } catch (error) {
-      console.error('Error creating wishlist:', error);
-    }
-  };
-
   useEffect(() => {
     const storedAddedStatus =
-      JSON.parse(localStorage.getItem(`addedStatus-${album._id}`)) || false;
+      JSON.parse(localStorage.getItem(`wishListAlbums-${album._id}`)) || false;
     setIsAdded(storedAddedStatus);
   }, [album._id]);
 
   const addToWishlist = async () => {
-    const wishlistId = await fetchOrCreateWishlist(); // Fetch the wishlist ID
     try {
       await axiosInstance.patch(
-        `${wishlistEndpoint}/${wishlistId}/add_album/${album._id}`
+        `${wishlistEndpoint}/${wishListId}/add_album/${album._id}`
       );
-      localStorage.setItem(`addedStatus-${album._id}`, JSON.stringify(true));
+      localStorage.setItem(`wishListAlbums-${album._id}`, JSON.stringify(true));
       setIsAdded(true);
     } catch (error) {
       console.error('Error adding album to wishlist:', error);
     }
   };
 
   const removeFromWishlist = async () => {
-    const wishlistId = await fetchOrCreateWishlist(); // Fetch the wishlist ID
     try {
       await axiosInstance.patch(
-        `${wishlistEndpoint}/${wishlistId}/remove_album/${album._id}`
+        `${wishlistEndpoint}/${wishListId}/remove_album/${album._id}`
+      );
+      localStorage.setItem(
+        `wishListAlbums-${album._id}`,
+        JSON.stringify(false)
       );
```

Comment:
If you end up making the localstorage field an object, this will need to be updated to take that into account:

```suggestion
      const wishlistAlbums = JSON.parse(localStorage.getItem('wishlistAlbums'))
      delete wishlistAlbums[album._id];
      localStorage.setItem("wishlistAlbums", JSON.stringify(wishlistAlbums));
```

Reply:
You may want to have the page either manually reload (with react-router this can be done with `const navigate = useNavigate();` and then where you wan to refresh `navigate(0);`. 

This will ensure that when you remove an item from the wishlist, it goes away from the wishlist view.

Alternatively, you could pass down an `onRemove` function from the `WishlistView` (it would have to be drilled through the AlbumGrid and Album components) that will tell the WishlistView, which album id was removed, and then in WishlistView we can call `setAlbums` to update the list of albums to be shown on the grid (the actual API call to remove from wishlist would still remain here in the button component). 

A third possible option for doing this is using a Redux action so that you don't have to drill the prop through multiple components.